### PR TITLE
More linting for requirements

### PR DESCRIPTION
**Pull request by bernt-matthias**
- check for missing version
- check for missing name

see https://github.com/galaxyproject/tools-iuc/pull/3978#discussion_r712919797

## How to test the changes?
(Select all options that apply)
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

File: lib/galaxy/tool_util/linters/general.py [hunk 1]
```diff
@@ -63,8 +63,12 @@ def lint_general(tool_source, lint_ctx):
 
     requirements, containers = tool_source.parse_requirements_and_containers()
     for r in requirements:
+        if r.name == '':
```
Comment (jmchilton) [inline lib/galaxy/tool_util/linters/general.py]: Maybe constraint these checks to package requirements?

File: test/unit/tool_util/test_tool_linters_general.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
+from galaxy.tool_util.linters import general
```
Comment (nsoranzo) [inline test/unit/tool_util/test_tool_linters_general.py]: I'm not sure if we should have `test_*.py` files that don't actually contain tests.

File: lib/galaxy/tool_util/linters/general.py (hunk 1 above)
Comment (nsoranzo) [reply lib/galaxy/tool_util/linters/general.py]: And we should also lint the `type` attribute, I think.

Comment (bernt-matthias) [reply lib/galaxy/tool_util/linters/general.py]: > Maybe constraint these checks to package requirements?

True.

> And we should also lint the type attribute, I think.

I think this should be covered by the xsd https://github.com/galaxyproject/galaxy/blob/b793ccdb463fef14e54a62fe8430807b96e8875b/lib/galaxy/tool_util/xsd/galaxy.xsd#L6076

File: test/unit/tool_util/test_tool_linters_general.py (hunk 2 above)
Comment (bernt-matthias) [reply test/unit/tool_util/test_tool_linters_general.py]: Should I rename the files or just put them back into one. The file seemed to big for me ..

File: lib/galaxy/tool_util/linters/general.py [hunk 3]
```diff
@@ -63,10 +63,15 @@ def lint_general(tool_source, lint_ctx):
 
     requirements, containers = tool_source.parse_requirements_and_containers()
     for r in requirements:
-        # Warn requirement attributes with leading/trailing whitespace:
-        if r.version != r.version.strip():
-            lint_ctx.warn(
-                WARN_WHITESPACE_MSG % ('Requirement version', r.version))
+        if r.type == "package":
+            if r.name == '':
```
Comment (nsoranzo) [inline lib/galaxy/tool_util/linters/general.py]:
```suggestion
            if not r.name:
```

File: lib/galaxy/tool_util/linters/general.py [hunk 4]
```diff
@@ -63,10 +63,15 @@ def lint_general(tool_source, lint_ctx):
...
+                lint_ctx.error("Requirement without name found")
+            if r.version is None or r.version == '':
```
Comment (nsoranzo) [inline lib/galaxy/tool_util/linters/general.py]:
```suggestion
            if not r.version:
```

File: lib/galaxy/tool_util/linters/general.py (hunk 1 above)
Comment (nsoranzo) [reply lib/galaxy/tool_util/linters/general.py]: > I think this should be covered by the xsd 

Correct, sorry for the noise!

File: test/unit/tool_util/test_tool_linters_general.py (hunk 2 above)
Comment (nsoranzo) [reply test/unit/tool_util/test_tool_linters_general.py]: It doesn't seem to be particularly large compared to the other unit test files.

**Review by nsoranzo — APPROVED** [review]

**Pull request merged**